SP2-91 create csv controller class (#25)
SDU-Semester-Project-2/Heatington: Documentation/Heatington/Controllers/Serializers/CsvController.md

Side note (as it is not denoted in the diagram): Upon entering a state the next character of the string is being examined.

### Serialize
`public static string Serialize(CsvData data, bool includeHeaderIfNotNull = true)`

data: the data to be serialized into CSV format<br>
includeHeaderIfNotNull: whether to include the header in the top row of the serialized data if it exists<br>
Returns a string with the serialized data.

Serializes data based on the CSV format defined in RFC4180.<br>
Only uses `"` characters when needed.

## CsvData
A class that holds a deserialized form of CSV data. That makes it possible to read your deserialized data as plain text or to convert it into a list of objects fit to hold that data.

### Properties

`public List<string[]> Table { get; private set; }`

`public string[]? Header { get; private set; }`

Table: holds the CSV data as plain text organized a list of arrays.<br>
Header: holds the header as plain text in an array.

### Constructor

`public CsvData(List<string[]> data, string[]? header = null)`

data: a table of deserialized data as plain text<br>
header: holds entries of the CSV header

Methods:

> ### Create\<T\>

Reviewer: I may be wrong, but I do believe this format is better for readabilty. 
```suggestion
### Create\<T\>

`public static CsvData Create<T>(List<T> data, string[]? header = null)`

Parameters:

- `data` (List<T>): this is a list of objects that you want to represent as CSV data. 
- `header` (string[]?, optional): a default header to be used. Default is null.

Returns:

- CsvData: Returns the `CsvData` created from `data` and `header`.
```

> ### Create\<T\>
> 
> `public static CsvData Create<T>(List<T> data, string[]? header = null)`
> 
> data: a list of objects to be represented as CSV data<br>
> header: a default header to be used<br>
> Returns the CsvData created from data and header
> 
> If header is null a header is created from the list of public properties of the T type. The CSV data will contain the values of the public properties of the T type turned to string via .ToString(), in case a property is null an empty string will be used as an entry.

Reviewer: I may be wrong, but I do believe this format is better for readabilty. The same is for another methods in this documentation.
```suggestion
### Create\<T\>

`public static CsvData Create<T>(List<T> data, string[]? header = null)`

Parameters:

- `data` (List<T>): this is a list of objects that you want to represent as CSV data. 
- `header` (string[]?, optional): a default header to be used. Default is null.

Returns:

- CsvData: Returns the `CsvData` created from `data` and `header`.
```

```
